src/parseWithPointers.ts

```diff
@@ -105,15 +109,16 @@ export const walkAST = (
         return container;
       }
       case Kind.SEQ:
-        return (node as YAMLSequence).items.map(item => walkAST(item, options, duplicatedMappingKeys));
+        return node.items.map(item => walkAST(item, options, duplicatedMappingKeys));
       case Kind.SCALAR:
-        return getScalarValue(node as YAMLScalar);
-      case Kind.ANCHOR_REF:
-        if (node.value !== void 0 && isCircularAnchorRef(node as YAMLAnchorReference)) {
-          node.value = dereferenceAnchor(node.value, (node as YAMLAnchorReference).referencesAnchor);
+        return getScalarValue(node);
+      case Kind.ANCHOR_REF: {
+        if (isObject(node.value) && isCircularAnchorRef(node)) {
+          node.value = (dereferenceAnchor(node.value, node.referencesAnchor) as YAMLNode) || void 0;
```

Comment:
A change in the casted type is kind of dangerous? What's up?

Reply:
Yeah, it's dangerous. I'm tricking the compiler a bit her, it might return either a node or null, but certainly not an array.
It also kind of sucks, as one time we have `undefined` value, and the other time we deal with `null` - this is mostly because of AST and how the values are represented. I'm afraid there is not much I can do about it, but would need to double-check it, as I might be wrong.

Reply:
Nice, all I wanted is to make you lose time and since you're going to look into this, my mission is accomplished :trollface:

Reply:
@XVincentX 
![image](https://user-images.githubusercontent.com/9273484/65976206-14ec2080-e470-11e9-8aa0-8aa9018ca8eb.png)
![image](https://user-images.githubusercontent.com/9273484/65976252-246b6980-e470-11e9-9fc9-19cbeb774c47.png)

:man_shrugging:

Reply:
But actually null makes sense in case of Mapping, and undefined makes sense in case of ANCHOR_REF as well.

Reply:
I'll actually create a follow-up PR for that in a minute - gonna be a bug-fix